release/authorization/README.md

```diff
@@ -0,0 +1,420 @@
+*   a defined KEY over a wildcard element in a keyed path.
+
+Authorization rules must be defined such that a single best match is possible.
+If the result of policy evaluation is more than one match, an error must be
+raised.
+
+Match rules permit a match against:
+
+*   User or Group (not both)
+*   an gNMI gNMI path
+*   an access method (READ / WRITE / SUBSCRIBE)
+
+An implicit deny is assumed, if there is no matching rule in the policy. Logging
+may be specified on a per-policy-rule basis as well as a default for the whole
+authorization policy.
+
+As a request is evaluated against the configured policy, a READ / SUBSCRIBE
+request for the configuration tree may traverse all of the tree and subtrees.
+For portions of the tree for which the user has no access no data will be
+returned. A WRITE request which attempts to write to a denied gNMI path or
+element will return an error to the caller.
+
+[gNMI paths](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#222-paths)
+are hierarchical, and rooted at a defined "origin". gNMOpenConfigI may contain paths
+such as:
+
+```proto
+    /a/b/c/d
+```
+
+Paths may also have attributes associated with the path elements such as:
+
+```proto
+    /a/b[key=foo]/c/d
+```
+
+Attributes may be wildcarded in the policy, such as:
+
+```proto
+    /a/b[key=*]/c/d
+```
+
+Wildcards are only valid in policy rules when applied to attributes.
+Permitted use:
+
+```proto
+    /a/b[key=*]/c/d
+```
+
+Not permitted use:
+```proto
+    /a/b[key=foo]/*/d
+```
+
+An example of two rules with similar paths that differ only with respect
+to attribute wildcarding:
+
+```proto
+   /a/b[key=FOO]/c/d
+   /a/b[key=*]/c/d
+```
+
+The first rule specifies a single key, that rule is more specific than
+the second rule, which specifies 'any value is accepted'.
+
+
+## Bootstrap / Install Options
+
+System bootstrap, or install, operations may include an authorization policy
+delivered during bootstrap operations. It is suggested that the bootstrap
+process include the complete authorization policy so all production tools
+and services have immediate authorized access to finish installation and
+move devices into production in a timely manner.
+
+Using the Secure Zero Touch Provisioning (sZTP - RFC8572) process for
+bootstrap/installation provides a method to accomplish this delivery, and
+the delivery of all other bootstrap artifacts in a secure manner.
+
+## Conflict Resolution
+
+Policy evaluation should end with a single best match policy for the provided
+user / path. If there is more than one 'best match', an error must be logged
+and evaluation should return a failure.
+
+## An Example Authorization Protobuf
+
+```proto
+# proto-file: //experimental/users/morrowc/gnXi/proto/authorization.proto
+# proto-message: Policies
+version: "UUID-1234-123123-123123"
+created_on: 1234567890
+# Define 2 groups.
+group {
+  name: "family-group"
+  members { name: "stevie" }
```

Comment:
it seems more consistent with the docs above to use `user` as the keyword here instead of `members` (or `group` if composing groups of users).  my 2c